catalog: minor improvements to system columns

RaduBerinde opened the pull request:
This change cleans up the definition of system column IDs and uses the
column ID to determine the SystemColumnKind more efficiently.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/75426)
<!-- Reviewable:end -->

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

rebase for the fix for the CI flake

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75426#-:-Mu9GIqmDt9uU87N3nhZ:bs9cc65)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
catalog: minor improvements to system columns

This change cleans up the definition of system column IDs and uses the
column ID to determine the SystemColumnKind more efficiently.

Release note: None

RaduBerinde commented:
bors r+
